### ISSUE TITLE
pluma 1.20.1 broken due libpeas changes

### ISSUE THREAD

**Issue by rosset** [mate-desktop/pluma#305]
libpeas on Fedora only supports python3, so for now pluma is broken

#### Link to downstream report of your Distribution
https://bugzilla.redhat.com/show_bug.cgi?id=1563868

commits:
https://src.fedoraproject.org/rpms/libpeas/c/5ecfd42cfa64ce0cc0650aabda126b21cc563805?branch=master

Error (fedora rawhide, development branch, will become Fedora 29)
sudo dnf install pluma
conflicting requests
  - nothing provides libpeas-loader-python needed by pluma-1.20.1-1.fc29.i686
  - nothing provides libpeas-loader-python needed by pluma-1.20.1-1.fc29.x86_64

**Comment by raveit65** [comment]
test build without internal python plugins.
https://koji.fedoraproject.org/koji/taskinfo?taskID=26471696
I created a branch https://github.com/mate-desktop/pluma/commits/disable-libpeas-plugins
with this commit
https://github.com/mate-desktop/pluma/commit/59054bb7dfbd5f6f985ddd1712da22262425b8a6
Maybe someone has the same pain......

**Comment by hadess** [comment]
Pluma, Eye of MATE and gtranslator are the only 3 things in Fedora depending on that plugin. You should port those to Python3, gedit and eog were already ported around GNOME 3.0 times.

**Comment by rosset** [comment]
> test build without internal python plugins.
> https://koji.fedoraproject.org/koji/taskinfo?taskID=26471696
> I created a branch https://github.com/mate-desktop/pluma/commits/disable-libpeas-plugins
> with this commit
> [59054bb](https://github.com/mate-desktop/pluma/commit/59054bb7dfbd5f6f985ddd1712da22262425b8a6)
> Maybe someone has the same pain......

thank you, this build is fine, but atm I moved to geany as my default editor.

**Comment by andrewfowlie** [comment]
@raveit65 I like pluma & python - I could help with some of the development of Python parts, possibly Python3 port if that's required

**Comment by raveit65** [comment]
That would be great.
Pluma needs to run with libpeas-loader-python3.

**Comment by stuaxo** [comment]
What's involved in porting, anybody got links to the patches that were made to gedit ?

I've got a plugin I'm getting to work in Gedit and Xed, I'd like to get Pluma too if possible.

**Comment by alexander255** [comment]
@stuaxo: I've started this some time ago, but lost interest. Here's the changes I made:
https://github.com/alexander255/pluma/tree/python3 (basically update peas loader + python `2to3` tool).
However I just found this branch from @monsta, which looks like higher quality work than what I did: https://github.com/mate-desktop/pluma/commits/python3
So if you're still interested that's helping port the remaining plugins on that branch is probably the way to go.
